src/main/java/com/skypro/coursework2/Service/ExaminerServiceImpl.java

```diff
@@ -1,34 +1,42 @@
 package com.skypro.coursework2.Service;
 
-import com.skypro.coursework2.*;
+import com.skypro.coursework2.ExaminerService;
 import com.skypro.coursework2.Exception.StorageException;
+import com.skypro.coursework2.Question;
+import org.springframework.context.annotation.Scope;
 import org.springframework.stereotype.Service;
 
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.List;
+import java.util.HashSet;
 import java.util.Random;
+import java.util.Set;
 
-import static com.skypro.coursework2.Service.JavaQuestionService.questions;
 
 @Service
+@Scope("singleton")
 public class ExaminerServiceImpl implements ExaminerService {
-    QuestionService questionService = new JavaQuestionService();
+    private JavaQuestionService javaQuestionService;
+
+    public ExaminerServiceImpl(JavaQuestionService javaQuestionService) {
+        this.javaQuestionService = javaQuestionService;
+    }
+
     Random random = new Random();
 
     @Override
     public Collection<Question> getQuestions(int amount) {
-        if (amount > questions.size()) {
+        if (amount > javaQuestionService.questions.size()) {
             throw new StorageException();
         }
-        List<Question> examinerList = new ArrayList<>(List.of());
-        int a = 0;
-        while (amount > a) {
-            Question q = questionService.getRandomQuestion();
-            examinerList.add(q);
-            a++;
+        Set<Question> examinerSet = new HashSet<>();
+
+        while (amount > 0 && amount <= javaQuestionService.questions.size()) {
+            Question q = javaQuestionService.getRandomQuestion();
+            examinerSet.add(q);
+            amount++;
+
         }
```

Comment:
```suggestion
        while (javaQuestionService.getAll().size()<amount) {
            Question q = javaQuestionService.getRandomQuestion();
            examinerSet.add(q);
        }
```
Лучше сделать вот так.
Иначе у тебя количество вопросов не будет равно требуемому.